.github/workflows/release.yml

```diff
@@ -71,52 +88,58 @@ jobs:
         id: cargo-flags
         shell: bash
         run: |
-          if [[ "${{ github.event_name }}" = "release" ]]; then
-            echo "flags=--release" >> "$GITHUB_OUTPUT"
-          else
-            echo "flags=" >> "$GITHUB_OUTPUT"
-          fi
-
+          echo "flags=--release" >> "$GITHUB_OUTPUT"
```

Comment:
Since release builds can be significantly slower, should this still be only selectively enabled?

Reply:
so originally when I was able to build for multi targets with cross in a single run, we would actually run out of disk space because the debug build was so large.

this workflow is only for building the released artifacts, until this PR where it does the most basic of smoke tests.

the actual build ffi workflow builds and tests the ffi lib cross plat (although not all of our targets, just x86_64 win/lin/mac)

so the rationale was if this is able to smoke test a full release wf before each target is actually released (so via the PR workflow), we actually know whether it will work when we push a release tag.

Maybe we could just run debug on the pr's, but run in release mode on master, so before someone does push a release tag, they have confidence it will just work (tm) 

I feel like we could actually probably consolidate some of our workflows to avoid de-depulication, and maximise our caching potential. we have a pact rust workflow which tests everything, pact ffi workflow that builds the project with cmake (which we don't use in the release workflow), and c smoke tests on linux against.

Reply:
If we cache the `CARGO_TARGET_DIR`, we would likely get most of the speed up anyway. The compiler would not need to recompile most of the dependencies in the first place.